[RNMobile] Add OfflineStatus component to display network connectivity

derekblank opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
Adds a mobile component to display a visual indicator when working offline within the mobile editor. (Note: the functionality to subscribe to network status will be handled in a separate PR.) 

## Why?
Overall, to improve the user's experience when uploading media. 

Addresses:
* https://github.com/wordpress-mobile/gutenberg-mobile/issues/6405

## How?
Adds OfflineStatus component and updates Layout component to host it.

## Testing Instructions
Applying the test patch will display the component statically until it is wired to the connectivity subscription hook proposed in https://github.com/WordPress/gutenberg/pull/56609.

<details><summary>Test Diff</summary>

```diff
diff --git a/packages/block-editor/src/components/block-list/index.native.js b/packages/block-editor/src/components/block-list/index.native.js
index 810e23e4c1..b4d69aa7c7 100644
--- a/packages/block-editor/src/components/block-list/index.native.js
+++ b/packages/block-editor/src/components/block-list/index.native.js
@@ -30,6 +30,7 @@ import {
 import { BlockDraggableWrapper } from '../block-draggable';
 import { useEditorWrapperStyles } from '../../hooks/use-editor-wrapper-styles';
 import { store as blockEditorStore } from '../../store';
+import OfflineStatus from '../offline-status';
 
 const identity = ( x ) => x;
 
@@ -235,6 +236,7 @@ export default function BlockList( {
 			onLayout={ onLayout }
 			testID="block-list-wrapper"
 		>
+			{ __DEV__ && <OfflineStatus /> }
 			{ isRootList ? (
 				<BlockListProvider
 					value={ {

```

</details>


## Screenshots or screencast <!-- if applicable -->
<img width="360" alt="Screenshot 2023-11-29 at 4 43 17 pm" src="https://github.com/WordPress/gutenberg/assets/643285/6dff7c5c-8dd3-4767-92c9-e89c3b09e45d">

apeatling committed:
Style Revisions: Remove style revisions dropdown menu (#56454)

* Remove dropdown menu

* Make sure setEditorCanvasContainerView() calls are made when needed so that toggle status works correctly when viewing any other sections of global styles.

* Pre select the revisions count to prevent flashes of incorrect revision counts. Hat tip @ramonjd.

* Remove CSS no longer needed.

* Modify e2e test to reflect removal of menu item.

* Fix issue with style book closing when browsing blocks.

* Replace onClick with onBack to reduce confusion with screen header click action.

* Change ordering of `isRevisionsOpened` const.

jffng committed:
Font Library: fix fonts not displaying correctly (#55393)

* Add kebab casing and font family wrapping in sanitization function.

Remove commented code.

* Simplify how kebab casing and quote wrapping is done.

* Ensure incoming font is returned.

* Add kebab casing to slug in form data.

* Add tests for wpKebabCase

* remove redundant test

* test 2 different inputs on first wpKebabCase case

* Move formatting functions to utils.

* reduce the scope of the function

* fix the live preview (without reloading the page) of the just installed fonts in the editor

* format php

* Add test for format_font_family.

* Format php.

* Add more test cases and format.

* Remove accidental badKey.

* Format.

---------

Co-authored-by: Vicente Canales <vicente.canales@automattic.com>
Co-authored-by: Sarah Norris <1645628+mikachan@users.noreply.github.com>
Co-authored-by: Sarah Norris <sarah@sekai.co.uk>
Co-authored-by: Matias Benedetto <matias.benedetto@gmail.com>

fullofcaffeine committed:
(RichText)(Workaround)(17.1.x) Fallback to a string arg in `collapseWhiteSpace()` if `value` is not a string (#56570)

* (RichText)(Fix)(Workaround) Make sure value is always a string to avoid TypeErorrs

See the comments in this changeset for more info on why this was needed.

* Use a more localized fix that does not change the argument

gutenbergplugin committed:
Bump plugin version to 17.1.3

gutenbergplugin committed:
Update Changelog for 17.1.3

dmsnell committed:
HTML API: Backport updates from Core (#56578)

 - Fix typo in Tag Processor example comment.
 - Rewrite @todo comments.
 - Add support for new tags in HTML Processor:
    - ADDRESS, ARTICLE, ASIDE, CENTER, DETAILS, DIALOG,
    - DIR, DL, FIELDSET, FOOTER, HEADER, HGROUP, MAIN,
    - MENU, NAV, SEARCH, SECTION, SUMMARY

tellthemachines committed:
Add setting to disable custom content size controls (#56236)

* Add setting to disable custom content size controls

* Make sure theme setting overrides block setting.

* Revert "Make sure theme setting overrides block setting."

This reverts commit ad4b76c5ddc6fe915c756294d30158c05f98319f.

* Change setting name to `allowCustomContentAndWideSize`

juanmaguitar committed:
Docs: Fundamentals of Block Development - Registration of a block (#56334)

* Add block registration documentation

* Update block registration process

* Update block registration in client code

* updated doc

* Update block registration settings

* Grammar check

* Moved Registration of A block to fundamentals of block development

* Added maarkdown extension

* Delete docs/getting-started/fundamentals-block-development/registration-of-a-block

* Removed "key"

* clarification path block.json

* Fix block.json registration path

* server-side features require block server registration

* server-side features require block server registration

* Update docs/getting-started/fundamentals-block-development/registration-of-a-block.md

Co-authored-by: Ramon <ramonjd@users.noreply.github.com>

* Update docs/getting-started/fundamentals-block-development/registration-of-a-block.md

Co-authored-by: Ramon <ramonjd@users.noreply.github.com>

* Update docs/getting-started/fundamentals-block-development/registration-of-a-block.md

Co-authored-by: Ramon <ramonjd@users.noreply.github.com>

* Update docs/getting-started/fundamentals-block-development/registration-of-a-block.md

Co-authored-by: Ramon <ramonjd@users.noreply.github.com>

* Update docs/getting-started/fundamentals-block-development/registration-of-a-block.md

Co-authored-by: Ramon <ramonjd@users.noreply.github.com>

* Update docs/getting-started/fundamentals-block-development/registration-of-a-block.md

Co-authored-by: Ramon <ramonjd@users.noreply.github.com>

* Fix formatting in registration-of-a-block.md

* Added link to new page and some text adjustments

* Update block registration function and build process reference

---------

Co-authored-by: Ramon <ramonjd@users.noreply.github.com>

juanmaguitar committed:
Docs: Fundamentals of Block Development - Working with Javascript in the Block Editor (#56553)

* Add JavaScript build process documentation

* Update docs/getting-started/fundamentals-block-development/javascript-in-the-block-editor.md

Co-authored-by: Fabian Kägy <mail@fabian-kaegy.de>

---------

Co-authored-by: Fabian Kägy <mail@fabian-kaegy.de>

juanmaguitar committed:
Docs: Fundamentals of Block Development - File structure of a block (#56551)

* Add file structure documentation for custom blocks

* Update docs/getting-started/fundamentals-block-development/file-structure-of-a-block.md

Co-authored-by: Ryan Welcher <me@ryanwelcher.com>

* Removed not clear sentence

---------

Co-authored-by: Ryan Welcher <me@ryanwelcher.com>

t-hamano committed:
File block: Remove anchor tag when copy pasting to file name (#56508)

* File block: Remove anchor tag when copy pasting to file name

* Don't strip default button text

* eEtract removeAnchorTag() function

Mamaduka committed:
Migrate 'align hook' e2e tests to Playwright (#56480)

* Migrate 'align hook' e2e tests to Playwright
* Remove old test files
* Update generated alignment markup assertions

ellatrix committed:
Editor styles: scope without adding specificity (#56564)

jameskoster committed:
Data list view: Make filter row, table header, and pagination sticky (#56157)

Mamaduka committed:
Workflows: Update 'days-before-stale' for flaky test report issues (#56585)

t-hamano committed:
Gallery Block: Use styled scrollbars for image captions (#56252)

* Gallery Block: Use styled scrollbars for image captions

* Add 1px border

* Apply transparent white color to scrollbar

youknowriad committed:
Documentation: Add the transforms page to the create block tutorial of the platform docs (#56559)

Co-authored-by: Héctor <27339341+priethor@users.noreply.github.com>

ntsekouras committed:
Dataviews: Add confirmation step before deleting a page (#56504)

* Dataviews: Add confirmation step before deleting a page

* Update component docs for DataViews

---------

Co-authored-by: Miguel Fonseca <150562+mcsf@users.noreply.github.com>

jameskoster committed:
Update trash icon (#56569)

jorgefilipecosta committed:
Update: Refactor useAddedBy to use authorText and originalSource fields. (#56568)

matiasbenedetto committed:
WP_Theme_JSON_Gutenberg: Add nested indexed array schema sanitization  (#56447)

* Add GB specific resolver

* changing unset function

* adding nested validation

* rename function

* Revert "Add GB specific resolver"

This reverts commit 27c0f6f652cf7c69509074a65aebf00ec01d727e.

* removing not needed check

* removing not needed check

* Remove keys from tree that are not arrays when they are defined as arrays in schema

* remove key if it is empty

* adding tests

* adding function docs

* php format

* add comment to function

---------

Co-authored-by: hellofromtonya <tonya.mork@automattic.com>

Commit:
Mobile - Fix issue when backspacing in an empty Paragraph block (#56496)

* Bring changes from #55134 to the mobile code

* Mobile - RichText - Force focus when the block is selected but the textinput is not, for cases when merging blocks.

* Update Buttons integration test due to a change in the logic of the app where deleting the only button available does not remove the block

* Mobile - Heading block - Adds integration test for merging a Heading block with an empty Paragraph block

* Mobile - Paragraph block - Adds integration test to check that backspacing in an empty Paragraph block merges succesfully with the previous block and keeps the focus on the TextInput

* Mobile - RichText - Set selection values to be the last character position when merging and adds some comments to explain what is doing

* Mobile - Paragraph block test - Use focusTextInput to check the TextInput is in focused instead of checking for the fomatting toolbar button

* Rename shouldFocusTextInputAfterUpdate to shouldFocusTextInputAfterMerge

* Update CHANGELOG

alexandrebuffet committed:
Rich Text: Update the regex that checks the format of className in Format Types to allow underscores (#56599)

ellatrix committed:
Raw handling: fix block schema merging (#56558)

jorgefilipecosta committed:
Dataviews: All Templates: Add filters to template author. (#56338)

juanmaguitar committed:
Fundamentals block development - landing and first pages (#56584)

* Add fundamentals of block development
documentation

* Remove JavaScript how-to guides and update block development fundamentals landing

* Update block development fundamentals documentation

* Manifest and TOC

* Update block development documentation links

* Update block development documentation links

* restored how-to-guides javascript

* Update TOC

* update folder name of fundamentals section

ellatrix committed:
Blocks pkg: remove 'browser' dependencies (#56433)

DAreRodz committed:
Interactivity API: migration to the new `store()` API (#55459)

* Copy new `store()` implementation

* Enable TS for the interactivity API

* Update directive bind tests

* Update runtime with latest changes

* Replace wp-effect tests with wp-watch tests

* Update wp-body tests

* Update wp-class tests

* Update wp-context tests

* Update wp-init tests

* Fix wp-key assignment

* Update wp-key tests

* Update wp-on tests

* Update directive priorities tests

* Update directive slot tests

* Update wp-style tests

* Update wp-text tests

* Update negation operator tests

* Update router navigate tests

* Refactor tests for router regions

* Removed afterLoad tests

* Fix initial state tag id

* Update initial state tests

* Update tovdom-islands tests

* Update tovdom tests

* Replace store with initial state in PHP

* Fix wp prefixes in initial state

* Add types declaration to package.json so that TS works

* Add `wp-each` directive in the new store API

* Add workaround for store API SDR

* Modify children instead of replacing the element

* Migrate File block to the new store API

* Refactor Image block to use the new store API

* Update runtime

* Fix file namespace

* Finish migrating the Image block

* Update Navigation block

* Replace effect with watch in Image

* Fix namespace assignment

* Fix image state getter names

* Fix getters in navigation state

* Update Query block

* Fix directives for Navigation block

* Add missing store namespace

* Allow forward slashes in namespaces

* Migrate Search block

* Update Initial State and its tests

* Revert "Add `wp-each` directive in the new store API"

This reverts commit 51bf723490556445073f0874939a4e88429c9dff.

* Revert "Update Initial State and its tests"

This reverts commit 7e5580c79f13a8f3580c2f40fdbdce65a1acfc13.

* Revert "Replace store with initial state in PHP"

This reverts commit 519880f8cd06e527f6fd50a91725fa641e29f8ac.

* Revert "Add workaround for store API SDR"

This reverts commit 933c4aca997926cf01cbace7ba864b2e554be9ec.

* Fix phpcs errors

* Update Query phpunit tests

* Fix Query's Interactivity API context definition

* Fix action name in Navigation

* Make `double()` prop a getter

Co-authored-by: Luis Herranz <luisherranz@gmail.com>

* Fix `double()` prop definition

* Rename effects to callbacks in tests

* Rename effects to callbacks in blocks

* Rename effects to callbacks in unit tests

* Move callbacks at the end of the store definition

* Remove `layout-init` directive

* Add default args for `openMenu` and  `closeMenu`

* Change Interactivity Store script tag ID

---------

Co-authored-by: Michal Czaplinski <mmczaplinski@gmail.com>
Co-authored-by: Mario Santos <santosguillamot@gmail.com>
Co-authored-by: Luis Herranz <luisherranz@gmail.com>

juanmaguitar committed:
Fundamentals of Block Development - fix save definition (#56605)

ramonjd committed:
Core data revisions: extend support to other post types (#56353)

* Initial commit: add revisions support for templates

* Renames constant to be clearer about what these post types apply to.
Adds wp_block and wp_navigation

* Testing updating record.key in the action.

* Moving `revisionKey` to the entity config and use it in the action and resolvers

* Me: Pray, depart hence, thou knavish trifle! Thy utility be as absent as mirth in a grave, a jest without jesters. Begone, thou witless motley, for even a jest doth surpass thy meager worth!
Empty line: *whimper*

Mamaduka committed:
Migrate 'block directory' e2e tests to Playwright (#56593)

* Migrate 'block directory' e2e tests to Playwright

* Remove old test files

* Skip class locators.

Co-authored-by: Bart Kalisz <bartlomiej.kalisz@gmail.com>

* Simplify the last assertion

---------

Co-authored-by: Bart Kalisz <bartlomiej.kalisz@gmail.com>

derekblank committed:
Add OfflineStatus component to display network connectivity

ntsekouras committed:
Dataviews: Grid layout refinements (#56441)

* Dataviews: Grid layout refinements

* grid compact item actions

* Revert tooltips and update styles

* temp fix to not render empty values in grid

* linting issues

* do not render description in grid view if empty

* add theme's background color in previews empty space

---------

Co-authored-by: James Koster <james@jameskoster.co.uk>

derekblank committed:
Update CHANGELOG

ellatrix committed:
Performance: avoid calling postcss when not needed (#56601)

juanmaguitar committed:
Docs: Fundamentals of Block Development - block.json (#56435)

* Add block.json file

* Add block.json file for block registration

* Refactor block.json for improved performance and
development

* Clarification over groups of properties for block.json

* Update block.json with file paths and enable UI panels

* Better organization of subsections

* reorganized content of the page and added examples

* Update block-json.md with attributes diagram

* Update block.json with improved descriptions and formatting

* Remove unnecessary code related to `supports` in
block-json.md

* Update docs/getting-started/fundamentals-block-development/block-json.md

Co-authored-by: Ryan Welcher <me@ryanwelcher.com>

* Update attributes passed to React components and
save function in block-json.md

* Delete block-json.md file

* Add block.json file to TOC and index

---------

Co-authored-by: Ryan Welcher <me@ryanwelcher.com>

getdave committed:
Restore placeholder to Post Title field (#56580)

juanmaguitar committed:
Docs: Fundamentals of Block Development - The block wrapper (#56596)

* Add The Block Wrapper

* Refactor block wrapper and add useBlockProps hook

* Update block wrapper in the Block Editor

* Update block markup and wrappers

* Update block markup and wrapper attributes

* Update block wrapper attributes and add documentation

* Content added

* Fix server-side render definition for block
wrapper

* Refactor block wrapper markup

* Update docs/getting-started/fundamentals-block-development/the-block-wrapper.md

Co-authored-by: Ryan Welcher <me@ryanwelcher.com>

* Update docs/getting-started/fundamentals-block-development/the-block-wrapper.md

Co-authored-by: Ryan Welcher <me@ryanwelcher.com>

* Update docs/getting-started/fundamentals-block-development/the-block-wrapper.md

Co-authored-by: Ryan Welcher <me@ryanwelcher.com>

* Update docs/getting-started/fundamentals-block-development/the-block-wrapper.md

Co-authored-by: Ryan Welcher <me@ryanwelcher.com>

* Update docs/getting-started/fundamentals-block-development/the-block-wrapper.md

Co-authored-by: Ryan Welcher <me@ryanwelcher.com>

* Fix save function in block wrapper

* Refactor block wrapper markup to add attributes

* update folder name

* Add block-wrapper.md to toc.json

* Add link create-block

* fixed link Metadata in block.json

---------

Co-authored-by: Ryan Welcher <me@ryanwelcher.com>

DAreRodz committed:
Add missing changelog entry (#56611)

Mamaduka committed:
Migrate 'block icons' e2e tests to Playwright (#56610)

* Migrate 'block icons' e2e tests to Playwright

Co-authored-by: mimi <mimitips@gmail.com>

* Remove old test file

---------

Co-authored-by: mimi <mimitips@gmail.com>

DAreRodz committed:
Create block: Update `interactive-template` to the new `store()` API (#56613)

* Update templates to the new `store()` API

* Add changelog entry

* Use wp_unique_id

---------

Co-authored-by: Luis Herranz <luisherranz@gmail.com>

dcalhoun commented on packages/edit-post/src/components/layout/offline-icon.native.js:
I think it may be most appropriate to place this icon within `packages/icons/src/library/`. WDYT?

dcalhoun commented on packages/edit-post/src/components/layout/offline-status.native.js:
```diff
@@ -0,0 +1,53 @@
+		translateY.value = withTiming( 0, { duration: 100 } );
+	};
+
+	useEffect( () => {
+		slideIn();
+	}, [] );
```
Do we need to return a clean up function or it safe for this animation to be invoked if the component is un-mounted?

derekblank replied:
In testing, I didn't feel this animation actually added much value. I've removed it for now -- we could test re-introducing it when this component is wired up to the connection subscription hook, or seek design feedback.

dcalhoun commented on packages/edit-post/src/components/layout/offline-status.native.js:
I wonder if we should relocate this component elsewhere. I suppose it will only be used for `edit-post` for the native mobile editor, but I wonder if it is more appropriate elsewhere. My first thought is within `packages/editor/src/components/`, similar to `AutosaveMonitor`. 

WDYT?

derekblank replied:
I agree. Moved the files to the `editor` package.

dcalhoun commented on packages/edit-post/src/components/layout/style.native.scss:
```diff
@@ -24,3 +24,15 @@
 	left: 0;
 }
+
+.offlineStatusContainer {
+	background-color: $light-ultra-dim;
+	padding: $grid-unit;
+	justify-content: center;
+	flex-direction: row;
+}
+
+.offlineText {
+	padding-left: 3;
+	padding-top: 2;
+}
```
If we ultimately relocate these styles to its own component file, it may make sense to follow BEM selector naming as I believe that is our current project practice.

derekblank replied:
I wasn't aware that BEM selector naming was the current practice. Looking at other native scss files, it appears to be [quite a mix](https://github.com/WordPress/gutenberg/blob/7b1f24c60e7ee71f3391349acf58d14f0bd03ce1/packages/block-library/src/image/styles.native.scss#L43-L70). I'm not against using BEM however.

dcalhoun replied:
I'll share some context via a direct message.

dcalhoun commented on packages/react-native-editor/CHANGELOG.md:
```diff
@@ -11,6 +11,7 @@ For each user feature we should also add a importance categorization label  to i
 
 ## Unreleased
 -   [***] Fix issue when backspacing in an empty Paragraph block [#56496]
+-   [**] Add OfflineStatus component [#56627]
```
Given we use this change log to produce user-facing release notes, I might suggest focusing on the impact rather than the implementation details. 
```suggestion
-   [**] Editor displays user of network connection status when offline [#56627]
```

dcalhoun reviewed (commented):
Good progress! I did not test the code given the draft state, but provided initial feedback.

getdave committed:
Restore Post Title visual styles in Code View mode (#56582)

* Apply styles to match previous visuals

* Adjust CSS in response to feedback

jeryj committed:
Migrate Is Typing Test to Playwright (#56616)

To simplify, the the test method for typing in dropdowns from a toolbar was switched to use an existing block with a dropdown rather than add a dropdown with an input to the toolbar manually.

oandregal committed:
DataViews: document `view.layout` (#56637)

flootr committed:
scripts: update `jest-dev-server` to v9 (#56552)

* scripts: update `jest-dev-server` to v9

* pass `setupServer` response to `teardown`

* add changelog entry

ntsekouras committed:
Dataviews: Add first e2e tests (#56634)

* Dataviews: Add first e2e tests

* refactor tests and add `createTemplate` util

ciampo committed:
Text and Heading: improve docs around default values and truncation logic (#56518)

* Text and Heading: improve docs around default values and truncation logic

* CHANGELOG

* Improve docs, fix typos

ciampo committed:
FormToggle: refine animation (#56515)

* ToggleForm: refine animation

* CHANGELOG

* Use same transition duration for both color and position-based styles

megane9988 committed:
Fix label of columns inspector panel (#56647)

luisherranz committed:
Interactivity API: Use modules instead of scripts in the frontend (#56143)

* Bundle `@wordpress/interactivity` as an ES module

* Add the class with a basic API

* Make it work with the Navigation block

* Add versions

* Register `@wordpress/interactivity` module

* Add query and image blocks

* Add id with module identifier to the enqueued modules

* Add requried $usage for admin/frontend

* Avoid the use of enqueue to register modules

* Refactor versions

* Switch from `both` to `['admin', 'frontend']`

* Improve comments

* Add an optional static dependency graph

* Add static and dynamic dependencies in the server

* Add the file and search blocks

* Move $version out of $args to match wp_register_script

* Improve version logic

* Add polyfill

* Fix $version using its own arg in register calls

* Add unit tests

* Refactor tests, add test get import map

* Cleaning data

* Use gutenberg_url()

* Use wp_print_script_tag

* Fix DocBlock on get_import_map()

* Load navigation module only on Gutenberg plugin

* Load query module only on Gutenberg plugin

* Load search module only on Gutenberg plugin

* Load file module only on Gutenberg plugin

* Load image module only on Gutenberg plugin

* Make registration optional

* Improve navigation logic

* Remove unnecessary check

* Fix missing view file

* Don't print the import map if it's empty

* Load the importmap polyfill locally

* Move the es-modules-shims package to the top

* Use the public functions in the tests

* Update test to be more output oriented

* Remove we from comments.

* Start using modules in the interactivity e2e tests

* Update package-lock.json

---------

Co-authored-by: Carlos Bravo <carlos.bravo@automattic.com>

gutenbergplugin committed:
Bump plugin version to 17.2.0-rc.1

gutenbergplugin committed:
Update Changelog for 17.2.0-rc.1

gutenbergplugin committed:
Update changelog files

gutenbergplugin committed:
chore(release): publish

 - @wordpress/a11y@3.47.0
 - @wordpress/annotations@2.47.0
 - @wordpress/api-fetch@6.44.0
 - @wordpress/autop@3.47.0
 - @wordpress/babel-plugin-import-jsx-pragma@4.30.0
 - @wordpress/babel-plugin-makepot@5.31.0
 - @wordpress/babel-preset-default@7.31.0
 - @wordpress/base-styles@4.38.0
 - @wordpress/blob@3.47.0
 - @wordpress/block-directory@4.24.0
 - @wordpress/block-editor@12.15.0
 - @wordpress/block-library@8.24.0
 - @wordpress/block-serialization-default-parser@4.47.0
 - @wordpress/block-serialization-spec-parser@4.47.0
 - @wordpress/blocks@12.24.0
 - @wordpress/browserslist-config@5.30.0
 - @wordpress/commands@0.18.0
 - @wordpress/components@25.13.0
 - @wordpress/compose@6.24.0
 - @wordpress/core-commands@0.16.0
 - @wordpress/core-data@6.24.0
 - @wordpress/create-block@4.31.0
 - @wordpress/create-block-interactive-template@1.10.0
 - @wordpress/create-block-tutorial-template@3.1.0
 - @wordpress/customize-widgets@4.24.0
 - @wordpress/data@9.17.0
 - @wordpress/data-controls@3.16.0
 - @wordpress/date@4.47.0
 - @wordpress/dependency-extraction-webpack-plugin@4.30.0
 - @wordpress/deprecated@3.47.0
 - @wordpress/docgen@1.56.0
 - @wordpress/dom@3.47.0
 - @wordpress/dom-ready@3.47.0
 - @wordpress/e2e-test-utils@10.18.0
 - @wordpress/e2e-test-utils-playwright@0.15.0
 - @wordpress/e2e-tests@7.18.0
 - @wordpress/edit-post@7.24.0
 - @wordpress/edit-site@5.24.0
 - @wordpress/edit-widgets@5.24.0
 - @wordpress/editor@13.24.0
 - @wordpress/element@5.24.0
 - @wordpress/env@8.13.0
 - @wordpress/escape-html@2.47.0
 - @wordpress/eslint-plugin@17.4.0
 - @wordpress/format-library@4.24.0
 - @wordpress/hooks@3.47.0
 - @wordpress/html-entities@3.47.0
 - @wordpress/i18n@4.47.0
 - @wordpress/icons@9.38.0
 - @wordpress/interactivity@3.0.0
 - @wordpress/interface@5.24.0
 - @wordpress/is-shallow-equal@4.47.0
 - @wordpress/jest-console@7.18.0
 - @wordpress/jest-preset-default@11.18.0
 - @wordpress/jest-puppeteer-axe@6.18.0
 - @wordpress/keyboard-shortcuts@4.24.0
 - @wordpress/keycodes@3.47.0
 - @wordpress/lazy-import@1.34.0
 - @wordpress/list-reusable-blocks@4.24.0
 - @wordpress/media-utils@4.38.0
 - @wordpress/notices@4.15.0
 - @wordpress/npm-package-json-lint-config@4.32.0
 - @wordpress/nux@8.9.0
 - @wordpress/patterns@1.8.0
 - @wordpress/plugins@6.15.0
 - @wordpress/postcss-plugins-preset@4.31.0
 - @wordpress/postcss-themes@5.30.0
 - @wordpress/preferences@3.24.0
 - @wordpress/preferences-persistence@1.39.0
 - @wordpress/prettier-config@3.4.0
 - @wordpress/primitives@3.45.0
 - @wordpress/priority-queue@2.47.0
 - @wordpress/private-apis@0.29.0
 - @wordpress/project-management-automation@1.46.0
 - @wordpress/react-i18n@3.45.0
 - @wordpress/readable-js-assets-webpack-plugin@2.30.0
 - @wordpress/redux-routine@4.47.0
 - @wordpress/reusable-blocks@4.24.0
 - @wordpress/rich-text@6.24.0
 - @wordpress/router@0.16.0
 - @wordpress/scripts@26.18.0
 - @wordpress/server-side-render@4.24.0
 - @wordpress/shortcode@3.47.0
 - @wordpress/style-engine@1.30.0
 - @wordpress/stylelint-config@21.30.0
 - @wordpress/sync@0.9.0
 - @wordpress/token-list@2.47.0
 - @wordpress/undo-manager@0.7.0
 - @wordpress/url@3.48.0
 - @wordpress/viewport@5.24.0
 - @wordpress/warning@2.47.0
 - @wordpress/widgets@3.24.0
 - @wordpress/wordcount@3.47.0

juanmaguitar committed:
Docs: Fundamentals of Block Development ---- Small fixes for "Block wrapper" (#56651)

* Update block wrapper markup in server-side render definition

* Update block wrapper link in README.md

luisherranz committed:
Move the importmap to a folder (#56655)

derekblank committed:
Move InserterButton from components package to block-editor package (#56494)

* Move mobile InserterButton from components package to block-editor package

* Add @wordpress/primitives to block-editor package dependencies

* Update react-native-editor CHANGELOG

* Move SVG and Path imports from @wordpress/primitives to @wordpress/components

* Update components package CHANGELOG

* Update InserterButton import paths

jeherve committed:
Social Icons: add Gravatar service (#56544)

This introduces a new Social Link icon, for the Gravatar service.

Gravatar is used in other places within WordPress, and offers profile pages that one can use to list their presence on the web. It can be useful to link to that from your blog.

tellthemachines committed:
Fix input not showing when switching to "Fixed" width (#56660)

* Fix input not showing when switching to "Fixed" width

* Always render BlockWithChildLayoutStyles

* Render BlockWithChildLayoutStyles if layout is enabled

derekblank committed:
Update packages/react-native-editor/CHANGELOG.md

Co-authored-by: David Calhoun <github@davidcalhoun.me>

derekblank committed:
Move offline icon to icons library

derekblank committed:
Add cleanup function to OfflineStatus animation

andrewserong committed:
Editor: Fix issue where createBlock in block template caused list view collapse (#56666)

Co-authored-by: Riad Benguella <benguella@gmail.com>

oandregal committed:
DataViews: make the `mediaField` not hidable (#56643)

kmanijak committed:
Decode some characters if used in taxonomy name so it's displayed correctly in Query Loop filters (#50376)

* Decode ampersand if used in taxonomy name so it's displayed correctly in Query Loop filters

* Use more generic decoding function and in the displayTransform props of FormTokenField

oandregal committed:
DataViews: make `Actions` styles the same as any other column header (#56654)

WunderBart committed:
Scripts: Enable skipping Playwright browser installation (#56594)

Mamaduka committed:
Migrate 'editor multi entity saving' e2e tests to Playwright (#56670)

* Migrate 'editor multi entity saving' e2e tests to Playwright
* Remove old test file

Mamaduka committed:
Migrate 'inner-blocks-locking-all-embed' e2e tests to Playwright (#56673)

* Migrate 'inner-blocks-locking-all-embed' e2e tests to Playwright

* Remove old test file

jameskoster committed:
Align data view icon usage (#56602)

Commit:
Mobile Release v1.109.1 (#56644)

* Release script: Update react-native-editor version to 1.109.0

* Release script: Update CHANGELOG for version 1.109.0

* Release script: Update podfile

* Update `react-native-editor` changelog

* Update `react-native-editor` changelog

* Mobile - Fix issue when backspacing in an empty Paragraph block (#56496)

* Bring changes from #55134 to the mobile code

* Mobile - RichText - Force focus when the block is selected but the textinput is not, for cases when merging blocks.

* Update Buttons integration test due to a change in the logic of the app where deleting the only button available does not remove the block

* Mobile - Heading block - Adds integration test for merging a Heading block with an empty Paragraph block

* Mobile - Paragraph block - Adds integration test to check that backspacing in an empty Paragraph block merges succesfully with the previous block and keeps the focus on the TextInput

* Mobile - RichText - Set selection values to be the last character position when merging and adds some comments to explain what is doing

* Mobile - Paragraph block test - Use focusTextInput to check the TextInput is in focused instead of checking for the fomatting toolbar button

* Rename shouldFocusTextInputAfterUpdate to shouldFocusTextInputAfterMerge

* Update CHANGELOG

* Release script: Update react-native-editor version to 1.109.1

* Release script: Update CHANGELOG for version 1.109.1

* Release script: Update podfile

---------

Co-authored-by: Carlos Garcia <fluiddot@gmail.com>

oandregal committed:
DataViews: export the view components as defaults (#56677)

ellatrix committed:
setImmutably: don't clone all objects (#56612)

Mamaduka committed:
Migrate 'site editor export' e2e tests to Playwright (#56675)

* Migrate 'site editor export' e2e tests to Playwright
* Remove old test file
* Simplify
* Use query argument enabling edit mode

ajlende committed:
Fix order of typography sizes and families (#56659)

Mamaduka committed:
Editor: Avoid double parsing content in 'getSuggestedPostFormat' selector (#56679)

brookewp committed:
`QueryControls`: Add opt-in prop for 40px default size (#56576)

* `QueryControls`: Add opt-in prop for 40px default size

* Update RangeControl 40px size to be opt-in

* Update changelog

fluiddot committed:
[RNMobile] Fixes a crash on pasting MS Word list markup (#56653)

* Add polyfill for Element.prototype.remove

* Enable unit tests of `raw-handling` API filter `ms-list-converter`

* Update `react-native-editor` changelog

ciampo committed:
FormToggle: do not use "/" math operator (#56672)

* FormToggle: do not use "/" math operator

* Use math.div

* CHANGELOG

derekblank committed:
Remove OfflineStatus animation

derekblank committed:
Move OfflineStatus component to editor package

derekblank committed:
Remove OfflineStatus styles from layout native style file

dcalhoun commented on packages/react-native-editor/CHANGELOG.md:
```diff
@@ -11,6 +11,7 @@ For each user feature we should also add a importance categorization label  to i
 
 ## Unreleased
 -   [*] [internal] Move InserterButton from components package to block-editor package [#56494]
+-   [**] Editor displays user of network  status when offline [#56627]
```
There was a typo in my original suggestion. 
```suggestion
-   [**] Editor displays network status when offline [#56627]
```

dcalhoun reviewed (commented):
The latest changes look good. I noted a few things while testing. 

First, we likely need to implement dark mode support. Currently, the status text is illegible when dark mode is enabled. 

<details><summary>iOS Screenshot</summary> 

![Offline status is illegible when dark mode is enabled](https://github.com/WordPress/gutenberg/assets/438664/f2b4f18a-f132-4dee-a031-cffa3579709e)

</details> 

Second, after relocating `OfflineStatus` to the `editor` package, the testing instructions result in a cyclic dependency that throws an error. If we intend to use `OfflineStatus` in `block-editor` we may need to relocate `OfflineStatus` — possibly to the `block-editor`. 

The cyclic dependency from the testing instructions is noted in this diagram with a red arrow. 

![Cyclic dependency caused by the testing instructions](https://github.com/WordPress/gutenberg/assets/438664/f53ddc93-6e56-4285-8771-0eef7fc90684)


<details><summary>Error Details</summary>

```
 ERROR  Error: Cannot unlock an undefined object., js engine: hermes
    at Gutenberg (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.wordpress.gutenberg.development:120500:38)
    at RCTView
    at View (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.wordpress.gutenberg.development:48203:43)
    at RCTView
    at View (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.wordpress.gutenberg.development:48203:43)
    at AppContainer (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.wordpress.gutenberg.development:48081:36)
    at gutenberg(RootComponent) (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.wordpress.gutenberg.development:84470:28)
 ERROR  Error: Cannot unlock an undefined object., js engine: hermes
    at Gutenberg (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.wordpress.gutenberg.development:120500:38)
    at RCTView
    at View (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.wordpress.gutenberg.development:48203:43)
    at RCTView
    at View (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.wordpress.gutenberg.development:48203:43)
    at AppContainer (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.wordpress.gutenberg.development:48081:36)
    at gutenberg(RootComponent) (http://localhost:8081/index.bundle//&platform=ios&dev=true&minify=false&modulesOnly=false&runModule=true&app=org.wordpress.gutenberg.development:84470:28)
 ERROR  TypeError: Cannot read property 'default' of undefined

This error is located at:
    in Gutenberg
    in RCTView (created by View)
    in View (created by AppContainer)
    in RCTView (created by View)
    in View (created by AppContainer)
    in AppContainer
    in gutenberg(RootComponent), js engine: hermes
 ERROR  TypeError: Cannot read property 'default' of undefined

This error is located at:
    in Gutenberg
    in RCTView (created by View)
    in View (created by AppContainer)
    in RCTView (created by View)
    in View (created by AppContainer)
    in AppContainer
    in gutenberg(RootComponent), js engine: hermes
```

</details>

derekblank committed:
Update packages/react-native-editor/CHANGELOG.md

Co-authored-by: David Calhoun <github@davidcalhoun.me>

derekblank committed:
Move OfflineStatus component to block-editor package

derekblank committed:
Add style support for Dark Mode and use BEM scss syntax

derekblank commented:
> First, we likely need to implement dark mode support. Currently, the status text is illegible when dark mode is enabled.

Updated to include dark mode support. 👍 

> Second, after relocating OfflineStatus to the editor package, the testing instructions result in a cyclic dependency

I encountered this as well when testing, and agree that the block-editor package may be a better choice. I've moved the files there. With that change, `OfflineStatus` can be imported privately within `block-list` without the need to export it. Extrapolating this testing patch further, eventually we'd replace `__DEV__` with the inverted boolean from the `isNetworkConnected` hook:

```diff
diff --git a/packages/block-editor/src/components/block-list/index.native.js b/packages/block-editor/src/components/block-list/index.native.js
index 810e23e4c1..b4d69aa7c7 100644
--- a/packages/block-editor/src/components/block-list/index.native.js
+++ b/packages/block-editor/src/components/block-list/index.native.js
@@ -30,6 +30,7 @@ import {
 import { BlockDraggableWrapper } from '../block-draggable';
 import { useEditorWrapperStyles } from '../../hooks/use-editor-wrapper-styles';
 import { store as blockEditorStore } from '../../store';
+import OfflineStatus from '../offline-status';
 
 const identity = ( x ) => x;
 
@@ -235,6 +236,7 @@ export default function BlockList( {
 			onLayout={ onLayout }
 			testID="block-list-wrapper"
 		>
+			{ __DEV__ && <OfflineStatus /> }
 			{ isRootList ? (
 				<BlockListProvider
 					value={ {
```

derekblank commented:
> In this case I might build atop #56609. I would then open the PR against `feat/connectivity-status` so that the PR showcases only the additional changes. Once #56609 merges, I would use the following command to rebase this branch atop the latest `trunk` that includes the merged dependent changes [...]

I think this is a good approach. I think we can begin merging the work in #56609 and associated PRs. Once merged, we can follow the rebasing steps you've outlined, and update `block-list` to something like this to have the OfflineStatus component available behind a dev flag in order to merge this PR:

```
{ __DEV__ && ! isConnected && <OfflineStatus /> }
```

WDYT? Do you note any other code changes needed outside of waiting on this describe #56609 merge flow?

derekblank commented on packages/react-native-editor/CHANGELOG.md:
```diff
@@ -11,6 +11,7 @@ For each user feature we should also add a importance categorization label  to i
 
 ## Unreleased
 -   [*] [internal] Move InserterButton from components package to block-editor package [#56494]
+-   [**] Editor displays network status when offline [#56627]
```
If we were to merge this PR within the next week with this work behind a `__DEV__` flag, this may not be the most accurate release note as it would not display network status until connected to the connectivity hook. Do we have a precedent for release notes with dev-only changes? Perhaps just `[internal]`?

dcalhoun replied:
I believe we have generally merely excluded release notes until the user-facing features are present. I.e. we would add the release note in the PR removing the `__DEV__` flag. 

That said, changing this release note to be an `[internal]` note more focused on the implementation details would likely be fine as well, e.g. `[internal] Add OfflineStatus component for communicating network connection state`.

derekblank commented on packages/block-editor/src/components/offline-status/index.native.js:
```diff
@@ -0,0 +1,43 @@
+	);
+
+	const iconStyle = usePreferredColorSchemeStyle(
+		styles.icon,
```
Noting that the linter will throw an error if these are written in square-bracket syntax, e.g., `styles[ 'icon' ]`:

```
  21:11  error  ["container"] is better written in dot notation  dot-notation
  26:11  error  ["text"] is better written in dot notation       dot-notation
  31:11  error  ["icon"] is better written in dot notation       dot-notation
```

Perhaps a quirk of BEM semantics, but using dot-notation for light styles and square-bracket syntax for dark styles, unless you have any suggestions.

dcalhoun replied:
Your approach is expected and correct. For better or worse, we only use bracket notations for selectors that include hyphens.

dcalhoun reviewed (commented):
Thanks for making the changes. 🙇🏻 

I noticed the contrast between the offline banner and top navigation header differs when comparing light and dark mode. The former uses different background colors for each element, the latter uses the same background color. A small detail, but I was unsure if the divergence was intentional. 

<details><summary>Appearance Divergence</summary>

| Light Mode | Dark Mode |
| - | - |
| ![light-mode-uses-differing background-colors](https://github.com/WordPress/gutenberg/assets/438664/2477deef-7d92-4f15-8b84-c4126c050dee) | ![dark-mode-uses-identical-background-colors](https://github.com/WordPress/gutenberg/assets/438664/de1b0815-b6c8-46f6-b0e0-44bdb7180808) |

</details>

> I think this [stacking branches] is a good approach. I think we can begin merging the work in #56609 and associated PRs. 

To clarify, the example `git rebase --onto` command I shared is only relevant if we rebased this branch atop the dependent `feat/connectivity-status` _before_ it is merged into the `trunk` branch. 

<details><summary>Stacked Branches Details</summary>

1. `git switch rnmobile/offline-status`
1. `git pull --rebase origin feat/connectivity-status`
1. Integrate the `isConnected` hook into the `OfflineStatus` component.
1. `git push origin rnmobile/offline-status --force-with-lease`
1. Update thie PR to target `feat/connectivity-status` instead of `trunk`.
1. Merge `feat/connectivity-status` into `trunk`. Note: GitHub will now automatically update this PR to target `trunk` instead of `feat/connectivity-status`.
1. Temporarily restore the `feat/connected-status` branch in GitHub.
1. `git fetch`
1. `git rebase --onto origin/trunk feat/connected-status rnmobile/offline-status`
1. `git push origin rnmobile/offline-status --force-with-lease`

Steps 1–5 would've been done originally when starting this work, building atop the dependent branch. Steps 6–10 would be done to rebase this branch atop `trunk` after the merge.


Now that we are likely about to merge `feat/connectivity-status` into `trunk`, this approach is less relevant for this particular PR.

</details>

If your plan is to await merging `feat/connectivity-status` before rebasing, then we can simply rebase this branch on `trunk` after the merge and work to integrate the newly available `isConnected` hook into this branch. 

<details><summary>Await Dependent Work Merge Details</summary>

1. Merge `feat/connectivity-status` into `trunk`.
1. `git switch rnmobile/offline-status`
1. `git pull --rebase origin trunk`
1. Integrate the `isConnected` hook into the `OfflineStatus` component.
1. `git push origin rnmobile/offline-status --force-with-lease`

We can now review the changes in this PR with the `isConnected` hook integrated.

</details>

> [...] Once merged, we can follow the rebasing steps you've outlined, and update `block-list` to something like this to have the OfflineStatus component available behind a dev flag in order to merge this PR:
> 
> ```
> { __DEV__ && ! isConnected && <OfflineStatus /> }
> ```

My thought was that we would use `isConnected` Hook _within_ the `OfflineStatus` component, returning `null` when `isConnected` is `true`. This would keep the network state closer to the component that uses it. It might also avoid unnecessary re-renders of `block-list` — not sure. WDYT?

derekblank commented:
> My thought was that we would use `isConnected` Hook _within_ the `OfflineStatus` component, returning `null` when `isConnected` is `true`.

I agree that using `isConnected` within the OfflineStatus component would likely help avoid re-renders of block-list. I'll follow the steps outlined in Await Dependent Work Merge Details later to sync this work with the hook.

dmsnell commented:
Thought for wherever this is resurrected, but the browser API also exposes online/offline events. I'm not sure if this already incorporates that element, but there's no reason we need to limit the status indicator to mobile devices.
